mmedit/apis/restoration_video_inference.py

```diff
@@ -38,35 +38,8 @@ def restoration_video_inference(model, img_dir, window_size, start_idx,
 
     device = next(model.parameters()).device  # model device
 
-    # pipeline
-    test_pipeline = [
-        dict(
-            type='GenerateSegmentIndices',
-            interval_list=[1],
-            start_idx=start_idx,
-            filename_tmpl=filename_tmpl),
-        dict(
-            type='LoadImageFromFileList',
-            io_backend='disk',
-            key='lq',
-            channel_order='rgb'),
-        dict(type='RescaleToZeroOne', keys=['lq']),
-    ]
-
-    # add normalize if it is used in the model
-    for op in model.cfg.train_pipeline:
-        if op['type'] == 'Normalize':
-            if 'gt' in op['keys']:  # remove 'gt' in keys
-                op['keys'].remove('gt')
-            test_pipeline.append(op)
-
-    test_pipeline.extend([
-        dict(type='FramesToTensor', keys=['lq']),
-        dict(type='Collect', keys=['lq'], meta_keys=['lq_path', 'key'])
-    ])
-
     # build the data pipeline
-    test_pipeline = Compose(test_pipeline)
+    test_pipeline = Compose(model.cfg.demo_pipeline)
```

Comment:
This should fall back to test_pipeline if no demo_pipeline exist, and to val_pipeline if neither demo_pipeline nor test_pipeline exist.